float32[] current_pos
float32 current_orientation
string pfield_id
---
float32[] velocity_final